eng/pipelines/smoke-test.yml

```diff
@@ -15,6 +17,28 @@ jobs:
           JavaVersion: '1.11'
 
     steps:
+      - task: Maven@3
+        displayName: Use the latest package version from the dev feed
+        inputs:
+          mavenPomFile: $(PomFile)
+          goals: 'versions:use-latest-versions'
```

Comment:
Is there anything that make this actually filter to dev versions? We also publish the GA versions to our dev feed and so in some cases they might be the latest. Perhaps that is what we want but I want to make sure you aren't assuming they are always a dev version.

Reply:
If there is not a `dev.yyyymmdd` in the package version then skip validating that its date matches. This was done to accommodate the case of packages that are not yet built and released to the dev feed (e.g. `azure-cosmos`). 

This coincidentally happens to work with the scenario you just highlighted. 

Also in our discussion: we have a history of not successfully building and releasing all packages to the dev feed on a given night, this should perhaps result in a warning (turning the pipeline execution yellow in reports) instead of an outright failure that blocks running the smoke tests. 

https://github.com/Azure/azure-sdk-for-java/pull/7079/files#diff-0040e4d7813e3d0df709e8a15331ae40R34